script/setup.js

```diff
@@ -197,191 +234,296 @@ try {
 		],
 		[`["src/**/*.ts!", "script/**/*.js"]`, `"src/**/*.ts!"`, "./knip.jsonc"],
 	]) {
-		await replace({ files, from, to });
+		try {
+			await replace({ files, from, to });
+		} catch (error) {
+			s.stop(chalk.red`❌ Error hydrating package metadata locally.`);
+
+			throw new Error(`Failed to replace ${from} with ${to} in ${files}`, {
+				cause: error,
+			});
+		}
 	}
 
-	await fs.writeFile(
-		".all-contributorsrc",
-		prettier.format(
-			JSON.stringify({
-				...JSON.parse((await fs.readFile(".all-contributorsrc")).toString()),
-				projectName: repository,
-				projectOwner: owner,
-			}),
-			{ parser: "json" }
-		)
-	);
+	try {
+		await writeFile(
+			".all-contributorsrc",
+			prettier.format(
+				JSON.stringify({
+					...JSON.parse((await readFile(".all-contributorsrc")).toString()),
+					projectName: repository,
+					projectOwner: owner,
+				}),
+				{ parser: "json" }
+			)
+		);
+	} catch (error) {
+		s.stop(chalk.red`❌ Error hydrating package metadata locally.`);
 
-	const endOfReadmeNotice = [
-		``,
-		`<!-- You can remove this notice if you don't want it 🙂 no worries! -->`,
-		``,
-		`> 💙 This package is based on [@JoshuaKGoldberg](https://github.com/JoshuaKGoldberg)'s [template-typescript-node-package](https://github.com/JoshuaKGoldberg/template-typescript-node-package).`,
-		``,
-	].join(EOL);
+		throw new Error(
+			"Could not update to '.all-contributorsrc' with new repository details. ",
+			{ cause: error }
+		);
+	}
 
-	await fs.appendFile("./README.md", endOfReadmeNotice);
+	try {
+		const endOfReadmeNotice = [
+			``,
+			`<!-- You can remove this notice if you don't want it 🙂 no worries! -->`,
+			``,
+			`> 💙 This package is based on [@JoshuaKGoldberg](https://github.com/JoshuaKGoldberg)'s [template-typescript-node-package](https://github.com/JoshuaKGoldberg/template-typescript-node-package).`,
+			``,
+		].join(EOL);
+
+		await appendFile("./README.md", endOfReadmeNotice);
+	} catch (error) {
+		s.stop(chalk.red`❌ Error hydrating package metadata locally.`);
 
-	console.log(chalk.gray`✔️ Done.`);
+		throw new Error(
+			"Could not append template-typescript-node-package details to 'README.md'. ",
+			{ cause: error }
+		);
+	}
 
-	console.log();
-	console.log(chalk.gray`Clearing CHANGELOG.md...`);
+	s.stop(chalk.green`✅ Hydrated all the provided details.`);
 
-	await fs.writeFile(
-		"./CHANGELOG.md",
-		prettier.format(`# Changelog`, { parser: "markdown" })
-	);
+	s.start(`Clearing CHANGELOG.md...`);
 
-	console.log(chalk.gray`✔️ Done.`);
+	try {
+		await writeFile(
+			"./CHANGELOG.md",
+			prettier.format(`# Changelog`, { parser: "markdown" })
+		);
+	} catch (error) {
+		s.stop(chalk.red`❌ Error clearing CHANGELOG.md.`);
 
-	console.log();
-	console.log(chalk.gray`Generating all-contributors table in README.md...`);
+		throw new Error("Could not empty 'CHANGELOG.md'. ", { cause: error });
+	}
```

Comment:
[Refactor] There are a lot of similar chunks of code that roughly:

1. `s.start(...)`
2. `try`/`catch`
3. `s.stop(...)`

I'd request that a general helper function be made for them (in #351) to help standardize. It's a good change!